app/authorization/AuthProvider.scala

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2018 The Delphi Team.
+// See the LICENCE file distributed with this work for additional
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package authorization
+
+
+import pdi.jwt.{Jwt, JwtAlgorithm, JwtClaim}
+
+
+object AuthProvider  {
+
+  /** This method generates JWT token for registering Delphi-Management at the Instance-Registry
+    *
+    *
+    * @param validFor
+    * @return
+    */
+
+
+  def generateJwt(validFor: Long = 1): String = {
+    val jwtSecretKey = "changeme"// configuration.get[String]("play.http.secret.JWTkey")
+    val claim = JwtClaim()
+      .issuedNow
+      .expiresIn(validFor*1200)
```

Comment:
you generate a token with a finite life span here and never regenerate a new token if the current one gets invalid. this needs to be fixed or else the server has to be restarted every 1200 seconds ;)

Reply:
What do you recommend. A token should be valid till the server is running or should I set the session period to 30 minutes?

Reply:
I believe neither of these suggestions is the solution we desire. The idea behind a token, which expires after some time, is that the damage taken if the token is compromised is limited due to the limited validation time. Thus making it valid for an indefinite time is not a good idea.
Just extending the expiration time to 30 minutes won't do any good either, because after 30 minutes the scala play server won't be able to query any content from the instance registry since the token expires and is never refreshed ( at least from what I get from your code the token is generated once when the Instance Registry controller is created).

To actually solve the problem the scala play server just has to recreate a new token once the old one is about to expire (with some time to spare). Therefore you could just set an interval on the scala webserver to create and assign a new token after let's say 5 minutes and set the lifetime of a token to 6 minutes and we should be fine.

Reply:
Hey Jan. I tried scheduling the generation of token by using akka scheduler in 8c786eafff5c93a4df4399bc85a12b7e1586c0f0. Even when the token was generated after 5 minutes. There seem to be no way to reassign the variable. If the variable was reassigned within scheduler it was inaccessible from outside.
Now, a new token is generated when a any of the method is called. It is valid till 5 minutes and as any of the method is called again new token is generated. So connection is never broken even if there is no activity. I think this approach is safe and good to go.
Only problem is Scala code styling. If I would have initialised the variable globally and used it in all the methods. I wouldn't be able to achieve what is expected.

Reply:
hey @Ayybeeshafi 
please consider the following code
```
  object AuthProvider {

    var token = ""
     /** This method generates JWT token for registering Delphi-Management at the Instance-Registry
      *
      * @param validFor
      * @return
      */

     def generateJwt()(): String = {
      
	
			if(token == "" || token =="checkIfExpires") {
			  token = "new"
			}
      
      return token

    }
  }
  
  AuthProvider.generateJwt()
  AuthProvider.generateJwt()
  AuthProvider.generateJwt()
```

You can just use your AuthProvider object to store the token conveniently in one place and return the same token as long as it is valid. Once it has expired you can just reassign the variable in the object and return the new token.